src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionItemExtensions.cs

```diff
@@ -60,15 +65,33 @@ public static VSCompletionItem ToVSCompletionItem(this CompletionItem completion
                 Detail = completion.Detail,
                 Documentation = completion.Documentation,
                 FilterText = completion.FilterText,
-                InsertText = completion.FilterText,
+                InsertText = completion.InsertText,
                 InsertTextFormat = completion.InsertTextFormat,
                 Kind = completion.Kind,
                 Label = completion.Label,
                 Preselect = completion.Preselect,
                 SortText = completion.SortText,
                 Tags = completion.Tags,
                 TextEdit = completion.TextEdit,
+                VsCommitCharacters = vsCommitCharacters,
```

Comment:
When setting the VSCommitCharacters should we be `null`ing out the non-VS `CommitCharacters` property?

Reply:
I was operating under the assumption that since `CommitCharacters` is part of the actual LSP spec I should always provide it.

Reply:
Definitely a fair concern. I think if we determine that we're running in VS that we should `null` it out in order to remove the serialization overhead.